File: direct/platforms/ios/WeexDemo/Weex/Module/virtual/UDP.m
//
//  Handle.m
//  Inteligence
//
//  Created by zzzl on 2018/10/9.
//  Copyright © 2018年 zzzl. All rights reserved.
//

#import "UDP.h"
#import "WeexDemo-Swift.h"

@implementation UDP
@synthesize weexInstance;

WX_EXPORT_METHOD(@selector(send:param:callback:))
WX_EXPORT_METHOD(@selector(interrupt:))
WX_EXPORT_METHOD(@selector(interruptAll))

@end

File: direct/platforms/ios/WeexDemo/Weex/Module/implement/UDP.swift
//
//  UDPHandle.swift
//  Inteligence
//
//  Created by zzzl on 2018/10/9.
//  Copyright © 2018年 zzzl. All rights reserved.
//

import Foundation
import Gzip
import CryptoSwift

extension UDP {
    
    @objc func send(_ cmd: Int, param: [String: AnyHashable], callback: @escaping WXModuleKeepAliveCallback) {
        guard let order = Order(rawValue: UInt8(truncatingIfNeeded: cmd)) else { callback([:], false); return }
        
        print(msg: "\(order) send")
        UDPManage.share.viewController = weexInstance.viewController as? BaseViewController
        UDPManage.share.perform(order: order, param: param, call: callback)
    }
    
    @objc func interrupt(_ cmd: Int) {
        guard let order = Order(rawValue: UInt8(truncatingIfNeeded: cmd)) else { return }
        print(msg: "\(order) interrupt")
        
        UDPManage.share.viewController = weexInstance?.viewController as? BaseViewController
        UDPManage.share.perform(order: order, param: nil, call: { (_, _) in }, isInterrupt: true)
    }
    
    @objc func interruptAll() {
        UDPManage.share.viewController = weexInstance.viewController as? BaseViewController
        UDPManage.share.interruptAll()
    }
}
